assets/javascripts/discourse/services/chat.js

```diff
@@ -294,39 +301,43 @@ export default class Chat extends Service {
       });
       this.currentUser.set("chat_channel_tracking_state", {});
       ajax("/chat/chat_channels.json").then((channels) => {
-        this.setProperties({
-          publicChannels: A(
-            this.sortPublicChannels(
-              (channels.public_channels || []).map((channel) =>
-                this.processChannel(channel)
-              )
-            )
-          ),
-          // We don't need to sort direct message channels, as the channel list
-          // uses a computed property to keep them ordered by `last_message_sent_at`.
-          directMessageChannels: A(
-            this.sortDirectMessageChannels(
-              (channels.direct_message_channels || []).map((channel) =>
-                this.processChannel(channel)
-              )
-            )
-          ),
-          hasFetchedChannels: true,
-          loading: false,
-        });
-        const idToTitleMap = {};
-        this.allChannels.forEach((c) => {
-          idToTitleMap[c.id] = c.title;
-        });
-        this.set("idToTitleMap", idToTitleMap);
-        this.presenceChannel.subscribe(channels.global_presence_channel_state);
+        this._processChannels(channels);
         this.userChatChannelTrackingStateChanged();
         this.appEvents.trigger("chat:refresh-channels");
         resolve(this._channelObject());
       });
     });
   }
 
+  _processChannels(channels) {
+    this.setProperties({
+      publicChannels: A(
+        this.sortPublicChannels(
+          (channels.public_channels || []).map((channel) =>
+            this.processChannel(channel)
+          )
+        )
+      ),
+      // We don't need to sort direct message channels, as the channel list
```

Comment:
Is this comment relevant? Looks like we're sorting them by calling "sortDirectMessageChannels".

Reply:
Yes never quite understood this comment, I should watch this more closely some day